yarn-project/aztec.js/src/account/entrypoint/schnorr_stored_key_account_entrypoint.ts

```diff
@@ -0,0 +1,64 @@
+import { AztecAddress, FunctionData, GrumpkinPrivateKey, TxContext } from '@aztec/circuits.js';
+import { Schnorr } from '@aztec/circuits.js/barretenberg';
+import { ContractAbi, encodeArguments } from '@aztec/foundation/abi';
+import { DebugLogger, createDebugLogger } from '@aztec/foundation/log';
+import { FunctionCall, PackedArguments, TxExecutionRequest } from '@aztec/types';
+
+import EcdsaAccountContractAbi from '../../abis/ecdsa_account_contract.json' assert { type: 'json' };
+import { DEFAULT_CHAIN_ID, DEFAULT_VERSION } from '../../utils/defaults.js';
+import { buildPayload, hashPayload } from './entrypoint_payload.js';
+import { CreateTxRequestOpts, Entrypoint } from './index.js';
+
+/**
+ * Account contract implementation that keeps a Schnorr signing public key in storage, and is retrieved on
+ * every new request in order to validate the payload signature.
+ */
+export class SchnorrStoredKeyAccountEntrypoint implements Entrypoint {
```

Comment:
Do you think we can avoid the code duplication by having a more comprehensive `signer` parameter? What I'm thinking of is:
- Keep the original single `StoredKeyAccountEntrypoint`, but only take `address` and `signer` as arguments (not `privateKey`)
- In this context, `signer` is not a stateless function, but rather a closure that includes the private key, and has the type `(message: Buffer) => Buffer`.
- We initialise the Schnorr entrypoint with a closure that uses the private key to sign the message, for instance in the `yarn-project/aztec.js/src/account/contract/schnorr_account_contract.ts`:
```typescript
public async getEntrypoint({ address }: CompleteAddress, { chainId, version }: NodeInfo) {
  const schnorr = await Schnorr.new();
  const signer = (msg: Buffer) => schnorr.constructSignature(msg, this.signingPrivateKey);
  return new StoredKeyAccountEntrypoint(address, signer, chainId, version);
}
```
- Same for the ECDSA one

Reply:
Great idea. Addressed in https://github.com/AztecProtocol/aztec-packages/pull/1919/commits/18d6287a7ac99d1da9cad2321487e9a07061ff47. It looks much better now.